File: frontend/src/app/types/address.ts
export interface Address {
  id: number;
  region: string;
  city: string;
  branchNumber: string;
  branchType: string;
  street: string;
  phone: string;
  workingHoursWeekdays: string;
  workingHoursWeekend: string;
}

export interface CreateAddressDto {
  region: string;
  city: string;
  branchNumber: string;
  branchType: string;
  street: string;
  phone: string;
  workingHoursWeekdays: string;
  workingHoursWeekend: string;
}

export type UpdateAddressDto = CreateAddressDto;

export interface BulkImportResult {
  successCount: number;
  failureCount: number;
  errors: string[];
}

File: backend/src/AddressApp.Domain/Entities/Address.cs
namespace AddressApp.Domain.Entities
{
    public class Address
    {
        public int Id { get; set; } // Guid yerine int
        public string Region { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string BranchNumber { get; set; } = string.Empty;
        public string BranchType { get; set; } = string.Empty;
        public string Street { get; set; } = string.Empty; // streetAddress yerine street
        public string Phone { get; set; } = string.Empty;
        public string WorkingHoursWeekdays { get; set; } = string.Empty;
        public string WorkingHoursWeekend { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}